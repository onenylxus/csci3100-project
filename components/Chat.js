// Import
import React from 'react';
import { View, Text } from 'react-native';

// Export Create Chat
export default function Chat() {
  const DATA = [
    {
      title: '31th March, 2021',
      data: ['Nice to meet you!', 'Nice to meet you!'],
      sender: ['me', 'others'],
    },
    {
      title: '1st April, 2021',
      data: [
        'My name is Mandy',
        'I am studying Mathematics',
        'My name is Sophy',
        'I am studying Statistics',
      ],
      sender: ['me', 'me', 'others', 'others'],
    },
    {
      title: 'Yesterday',
      data: ['This is a long message XDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD', 'HAHA'],
      sender: ['me', 'others'],
    },
    {
      title: 'Today',
      data: ['Hi', 'hi'],
      sender: ['others', 'me'],
    },
  ];

  const message = [];

  function messagePosition(index1, index2, props) {
    if (DATA[index1].sender[index2] === 'me') {
      if (props === 'messagePos') {
        return 'flex-end';
      }
      if (props === 'messageColor') {
        return '#0036D3';
      }
      if (props === 'textColor') {
        return '#FFFFFF';
      }
    }
    if (props === 'messagePos') {
      return 'flex-start';
    }
    if (props === 'messageColor') {
      return '#D2D2D2';
    }
    if (props === 'textColor') {
      return '#000000';
    }
  }
  // eslint-disable-next-line no-plusplus
  for (let index1 = 0; index1 < DATA.length; index1++) {
    message.push(
      <View
        style={{
          alignSelf: 'center',
          textAlign: 'center',
          justifyContent: 'center',
          backgroundColor: '#D2D2D2',
          borderRadius: 15,
          marginVertical: 5,
        }}
      >
        <Text style={{ color: '#000', padding: 5 }}>{DATA[index1].title}</Text>
      </View>
    );
    // eslint-disable-next-line no-plusplus
    for (let index2 = 0; index2 < DATA[index1].data.length; index2++) {
      message.push(
        <View>
          <View
            style={{
              alignSelf: messagePosition(index1, index2, 'messagePos'),
              padding: 10,
              backgroundColor: messagePosition(index1, index2, 'messageColor'),
              margin: 5,
              maxWidth: '80%',
              borderRadius: 15,
            }}
          >
            <Text
              style={{ color: messagePosition(index1, index2, 'textColor') }}
            >
              {DATA[index1].data[index2]}
            </Text>
          </View>
        </View>
      );
    }
  }

  return <View>{message}</View>;
}
