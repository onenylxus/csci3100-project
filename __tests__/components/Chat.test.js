// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import Chat from '../../components/chat';

// Run jest tests
describe('Chat', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<Chat />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
