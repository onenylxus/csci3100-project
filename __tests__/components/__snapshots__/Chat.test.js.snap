// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat matches snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#d2d2d2",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "padding": 5,
        }
      }
    >
      31th March, 2021
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#0036D3",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        Nice to meet you!
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontSize": 11,
          }
        }
      >
        12:30
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#D2D2D2",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        Nice to meet you!
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#000000",
            "fontSize": 11,
          }
        }
      >
        12:32
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#d2d2d2",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "padding": 5,
        }
      }
    >
      1st April, 2021
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#0036D3",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        My name is Mandy
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontSize": 11,
          }
        }
      >
        14:01
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#0036D3",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        I am studying Mathematics
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontSize": 11,
          }
        }
      >
        14:44
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#D2D2D2",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        My name is Sophy
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#000000",
            "fontSize": 11,
          }
        }
      >
        14:44
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#D2D2D2",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        I am studying Statistics
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#000000",
            "fontSize": 11,
          }
        }
      >
        14:45
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#d2d2d2",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "padding": 5,
        }
      }
    >
      Yesterday
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#0036D3",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        This is a long message XDDDDDDDDDDDDDDDDDDDDDDDDDDD
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontSize": 11,
          }
        }
      >
        12:30
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#D2D2D2",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        HAHA
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#000000",
            "fontSize": 11,
          }
        }
      >
        12:32
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#d2d2d2",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "padding": 5,
        }
      }
    >
      Today
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#D2D2D2",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        Hi
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#000000",
            "fontSize": 11,
          }
        }
      >
        12:30
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#0036D3",
          "borderRadius": 15,
          "margin": 3,
          "maxWidth": "80%",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        hi
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#FFFFFF",
            "fontSize": 11,
          }
        }
      >
        12:32
      </Text>
    </View>
  </View>
</View>
`;
